espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ActivityHelpers.kt

```diff
@@ -87,6 +89,19 @@ object ActivityHelpers {
             AndroidLogger.logger.info("Staring activity with custom options: $intentOptions")
             makeIntent(intentOptions)
         }
-        instrumentation.startActivitySync(intent)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
I would rewrite this code in the following manner to keep the tests working:

```kotlin

         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && displayId != null) {
             val options = ActivityOptions.makeBasic()
             options.launchDisplayId = displayId.toInt()
             startActivitySync(intent, options)
         } else {
             instrumentation.startActivitySync(intent)
         }      

```

The `startActivitySync` could be simply copied from SDK and adapted for Android below version P if needed. private class members may be extracted with reflection from the actual instrumentation instance (we have a convenience wrappers for these in `ReflectionUtils` and `KReflectionUtils`)

Reply:
My device is API 27 which have no Instrumention.StartActivitySync(intent, options), so i have to try this way.

Reply:
How about this?
![image](https://user-images.githubusercontent.com/12910117/77042115-185f6e00-69f6-11ea-872c-feba4ba029e9.png)

Reply:
I assume that not using startActivitySync is failing the tests. For now this solution would suffice though to not make it overcomplicated. I assume the displayId option is not going to be used much

Reply:
Yes. But it can help me to start my cases because of our primary display id is 1.

Reply:
yes, that is why I say it is fine, so you could be unblocked with your test